processing/src/test/java/org/apache/druid/java/util/common/IntervalsTest.java

```diff
@@ -78,4 +79,16 @@ public void testFindOverlappingInterval()
     );
   }
 
+  @Test(expected = DruidException.class)
+  public void testInvalidInterval()
+  {
+    try {
+      Intervals.of("invalid string");
+    }
+    catch (DruidException e) {
+      Assert.assertTrue(e.getMessage().contains("Bad Interval[invalid string]"));
+
+      throw e;
+    }
+  }
```

Comment:
There is a `DruidExceptionMatcher` that can be used to validate `DruidException` classes.  We try to use it with `Assert.assertThat()`.

You can do an `Assert.assertThrows()` first which will return the Throwable, and then `Assert.assertThat` on the returned throwable to validate with the matcher.  This pattern is preferred to expecting an exception be thrown out of the test class as it's a bit more explicit and keeps any instance of an exception exiting the test method as a bad thing always.